### DISCUSSION TITLE
Random - DB - Things :+1:

### DISCUSSION THREAD

**Discussion by helio-frota** [guacsec/trustify#136] [General]
*  Apparently something is printing a binary
`RUST_LOG=debug cargo test -- ingest_sboms --nocapture`
![2024-04-04_08-42](https://github.com/trustification/trustify/assets/6443576/4f080ea8-0983-4dc7-a613-5ca80317a810)

* I've changed all the tests to `let db = Database::for_test("a").await?;` and it works :+1: 
![2024-04-04_08-49](https://github.com/trustification/trustify/assets/6443576/1b8f02e2-3584-4080-92e2-f05aa8bfa520)
![2024-04-04_08-48](https://github.com/trustification/trustify/assets/6443576/26a0e921-fa86-4f57-b647-0c5d64916796)

**UPDATE:** https://github.com/trustification/trustify/issues/141#issuecomment-2037392134 we have a risk when running tests in parallel

* I don't know the best place to use this

```rust
let mut pg_dump = PgDumpBuilder::from(postgresql.settings())
    .schema_only()
    .file("out.sql")
    .build();
let (_stdout, _stderr) = pg_dump.execute()?;
```
  * Currently `db.dump()` makes no sense, as best-place-to-use because `db` is seaORM DBConnection related
    * _:
    ```rust
    #[derive(Clone, Debug)]
    pub struct Database {
        pub db: DatabaseConnection,  <-----
        db_strategy: Arc<DbStrategy>,
    } 
    ```
  * warning: if we add this line `let pg_clone = postgresql.clone();` then is enough to get this:
![2024-04-04_08-55](https://github.com/trustification/trustify/assets/6443576/198bafa6-ac9d-4da5-a7f6-f29a553fdda8)

* Related to the dump usage (I'm not 100% sure) probably we will need something like:
```rust
#[tokio::test]
    async fn package_version_not_affected_assertions() -> Result<(), anyhow::Error> {
        let db = Database::for_test("a").await?;
        let system = Graph::new(db);

        // things doing things here...

        assert_eq!(assertions.assertions.len(), 1);

        foo.dump() < ---- call dump here ----

        Ok(())
    }
```

**Comment by helio-frota** [comment]
> I've changed all the tests to let db = Database::for_test("a").await?; and it works

I don't know the reason, maybe it could be simplified ? :shrug: 

> Database::for_test().await?;

**Comment by helio-frota** [comment]
Makes no sense to call `dump` for each test [?]

![2024-04-04_09-34](https://github.com/trustification/trustify/assets/6443576/cd047a29-8d3c-4339-ba3f-092627b1d836)

is **PM Mode** the best place to have a db dump ?

**Reply by jcrossley3** [reply]
I don't view them as related, no. I would expect an `out.sql` file to only be of interest to a developer.

**Reply by helio-frota** [reply]
Fully agree with the interest part :+1: 

extra context --> Currently [since we are creating multiple databases (using different database names)] we are going to end up with multiple file dumps, instead of one dump-file. That is why I mentioned the PM mode. :+1:

**Reply by jcrossley3** [reply]
What triggers the dump?

**Reply by helio-frota** [reply]
this block of code:

```rust
let mut pg_dump = PgDumpBuilder::from(postgresql.settings())
    .schema_only()
    .file("out.sql")
    .build();
let (_stdout, _stderr) = pg_dump.execute()?;
```

* We need to find a good way to get the `postgresql.settings()`
* We need to call that after the `boostrap` part :+1: 
* We need to update the dependency and add a new one

```toml
postgresql_embedded = { workspace = true, features = ["blocking", "bundled", "tokio"] }
postgresql_commands = { workspace = true }
```

**Comment by helio-frota** [comment]
thanks, we have this issue now 
https://github.com/trustification/trustify/issues/139
